test/orange/orangeinp/CsgTreeUtils.test.cc

```diff
@@ -385,6 +385,119 @@ TEST_F(CsgTreeUtilsTest, calc_surfaces)
     EXPECT_EQ((std::vector<S>{S{1}, S{3}}), calc_surfaces(tree_));
 }
 
+TEST_F(CsgTreeUtilsTest, transform_negated_joins)
+{
+    auto s0 = this->insert(Surface{S{0}});
+    auto s1 = this->insert(Surface{S{1}});
+    auto n0 = this->insert(Negated{s1});
+    auto j0 = this->insert(Joined{op_and, {s0, n0}});
+    auto simplified = transform_negated_joins(tree_);
+
+    // check well-formed tree
```

Comment:
For consistency with the rest of the code, please try to capitalize the comments and write them as a sentence without the trailing `.`, unless it's an adjective phrase describing data or something
```suggestion
    // Check a well-formed tree
```